### ISSUE TITLE
org_create_assignment errors on trying to mirror repo

### ISSUE THREAD

**Issue by merliseclyde** [rundel/ghclass#82]
Using a repo that is either a template or not, I am getting errors when trying to create an assignment. 

For the case that it is not a repo, it fails on trying to clone (repo_mirror) the repo with an error 

  unused argument (mirror = mirror)
Error: Unable to locate the repo(s): '/private/var/folders/n4/nj1122xj6bn5_xgbptv7bml40000gp/T/RtmpYRsjni/HW5'.

For a template repo it gives the error

✖ Failed to mirror 'STA521-F19/HW5' to 'STA521-F19/HW5-Team5-01'.
  GitHub API error (422): 422 Unprocessable Entity
  └─ Unprocessable Entity (See
     https://developer.github.com/v3/repos/#create-repository-using-a-repository-template)

**Comment by mine-cetinkaya-rundel** [comment]
I was having a similar issue, I think it has to do with the gert branch being merged. I got this error when running `org_create_assignment()`but I was able to get to the same result with doing the three steps individually: `repo_create()`, `repo_add_team()`, `repo_mirror()`. 

`repo_mirror_template()` also failed for me.

**Comment by merliseclyde** [comment]
Yes - I had installed rundel@master, but with mirror it is working.  

The repo_mirror step now asks for confirmation for every repo (or more); the repo was created prior, but failed to clone.   Can there be an overwrite=TRUE option ?

**Comment by merliseclyde** [comment]
Using a template repo still fails for me...

**Comment by rundel** [comment]
I'd think both of these issues should be fixed now.

The prompting was me just not handling the possibility of writing things gracefully - repo_mirror now checks the targets to see if they have any commits - if there are more than one then it will prompt you to rerun with `overwrite=TRUE`.

The template repo issue I believe was likely due to GitHub's template mirroring requiring that the target repo does not exist. Their process creates the repo and then populates it -  the error messaging and checks wrt were terrible so I've made some changes to improve that. 

Let me know if this solves things or if there are lingering issues.

**Comment by merliseclyde** [comment]
Thanks!!!   Will check it out with next assignment :-)
cheers,
m


Merlise A Clyde
Professor Department of Statistical Science
Duke University

clyde@duke.edu<mailto:clyde@duke.edu>
919 681 8440

On Oct 18, 2019, at 7:14 AM, Colin Rundel <notifications@github.com<mailto:notifications@github.com>> wrote:


I'd think both of these issues should be fixed now.

The prompting was me just not handling the possibility of writing things gracefully - repo_mirror now checks the targets to see if they have any commits - if there are more than one then it will prompt you to rerun with overwrite=TRUE.

The template repo issue I believe was likely due to GitHub's template mirroring requiring that the target repo does not exist. Their process creates the repo and then populates it - the error messaging and checks wrt were terrible so I've made some changes to improve that.

Let me know if this solves things or if there are lingering issues.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_rundel_ghclass_issues_82-3Femail-5Fsource-3Dnotifications-26email-5Ftoken-3DAAC54FHKLYIRBMQMONHQDFDQPGLCPA5CNFSM4JBTLP22YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBT6SMQ-23issuecomment-2D543680818&d=DwMCaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=x3pHRqWA0t5CXMKfReZn7ZqLj13kFVUgFfTldRCztdQ&m=uV5qkF_u5YTsTZ2xFUKPeSAgEK5woFTtUMtI2RhRzdY&s=o1ISFe7a3qvVuyrnVHxRD-j98F2B9lxjdj9zl2nQp5Q&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAC54FCCZVTR3MRA2BTBL2LQPGLCPANCNFSM4JBTLP2Q&d=DwMCaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=x3pHRqWA0t5CXMKfReZn7ZqLj13kFVUgFfTldRCztdQ&m=uV5qkF_u5YTsTZ2xFUKPeSAgEK5woFTtUMtI2RhRzdY&s=kqLFKlz8IXVgMpahDuY2xPuGu6D9StOCW4IpivoUg3I&e=>.



Merlise A Clyde
Professor Department of Statistical Science
Duke University
http://stat.duke.edu/~clyde

clyde@duke.edu<mailto:clyde@duke.edu>
919 681 8440

**Comment by rundel** [comment]
Mirror behaviour and template handling were reworked in subsequent releases. Closing as stale — please reopen if you still encounter this on the latest version.
